### PR TITLE
Patrick/inverse logic validation

### PR DESCRIPTION
**Pull request by patrickmcewen**

### CODE REVIEW

**Comment by patrickmcewen** [comment]
tracking issues (see added log.txt)
- EDP lower in 40nm than in 7nm
- although I haven't verified this, I think EDP is dominated by memory/buffer. So when I am changing the transistor for the tech params its not making much of a difference.

**Comment by fazerlicourice7** [comment]
oh lol we're fucked. for the same application and architecture?

**Comment by patrickmcewen** [comment]
> oh lol we're fucked. for the same application and architecture? 

using matmult with mm_test but transistor size overwritten to 40

**Comment by fazerlicourice7** [comment]
that's so weird

File: src/symbolic_simulate.py [hunk 1]
```diff
@@ -288,11 +288,13 @@ def calculate_edp(self, hw):
             hw, self.execution_time
         )
 
+        print(f"total passive energy: {self.total_passive_energy}", flush=True)
         self.edp = self.execution_time * (self.total_active_energy + self.total_passive_energy)
 
         self.execution_time = self.execution_time.xreplace(cacti_subs)
         self.total_active_energy = self.total_active_energy.xreplace(cacti_subs)
         self.total_passive_energy = self.total_passive_energy.xreplace(cacti_subs)
+        self.edp = self.execution_time * (self.total_active_energy + self.total_passive_energy)
```
Comment (fazerlicourice7) [inline src/symbolic_simulate.py:297]: Actually if you log things a bit closely. this value (after evaluating E_p, E_a and D and then multiplying) doesn't agree with the edp you get if you combine the expressions, and then evaluate. 
This is also something important to look at

**Pull request closed without merging**